test/sharness/t0700-remotepin.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+test_description="Test ipfs remote pinning operations"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+
+# create user on pinning service
+test_expect_success "creating test user on remote pinning service" '
+        echo CI host IP address ${CI_HOST_IP} &&
+        export TEST_PIN_SVC=http://${CI_HOST_IP}:5000/api/v1 &&
+        ipfs pin remote service add test_pin_svc $TEST_PIN_SVC $(curl -X POST $TEST_PIN_SVC/users -d email=sharness@ipfs.io | jq --raw-output .access_token)
+'
+
+test_remote_pins() {
+  BASE=$1
+  if [ -n "$BASE" ]; then
+    BASE_ARGS="--cid-base=$BASE"
+  fi
+
+  test_expect_success "create some hashes using base $BASE" '
+    export HASH_A=$(echo "A" | ipfs add $BASE_ARGS -q --pin=false) &&
+    export HASH_B=$(echo "B" | ipfs add $BASE_ARGS -q --pin=false) &&
+    export HASH_C=$(echo "C" | ipfs add $BASE_ARGS -q --pin=false)
+  '
+
+  test_expect_success "check connection to pinning service" '
+    ipfs pin remote ls --service=test_pin_svc --enc=json
+  '
+
+  test_expect_success "'ipfs pin remote add'" '
+    export ID_A=$(ipfs pin remote add --service=test_pin_svc --enc=json $BASE_ARGS --name=name_a $HASH_A | jq --raw-output .RequestID) &&
+    export ID_B=$(ipfs pin remote add --service=test_pin_svc --enc=json $BASE_ARGS --name=name_b $HASH_B | jq --raw-output .RequestID) &&
+    export ID_C=$(ipfs pin remote add --service=test_pin_svc --enc=json $BASE_ARGS --name=name_c $HASH_C | jq --raw-output .RequestID) &&
+    sleep 3 # provide time for the pinning service to download the file
```

Comment:
I think --background may not be the best design. Because now you don't have a non-flaky way of checking that a background add has succeeded. What works better is a new command 'ipfs pin remote wait request-id' which blocks until the pin succeeds/fails. And keep 'pin add' always in the background.

Reply:
I don't think we need a new command for this since it's simple to do at the layer above this. For example, the way the WebUI will use this is likely:

- `ipfs pin remote add bafyfile --background` and extract the request-ID from the JSON
- Periodically run `ipfs pin remote ls --id=request-ID` and check the status field to see if it's pinned

We can do this in sharness too and it's not too rough (a for loop + jq)

Reply:
Sounds good. Incidentally, the http client does not have "ls by request id", but this is not crucial. I can use name.

Reply:
We might need to fix that to help out the WebUI. I think there's a GetStatus function that takes a single request-id that should help for now though